snapcraft/extensions/basic_graphics.py

```diff
@@ -0,0 +1,134 @@
+_EXT_NAME = "basic-graphics"
+
+
+class BasicGraphics(Extension):
+    """An extension that provides graphics libraries for snaps that don't need to integrate with a full desktop environment.
+
+    This extension is intended to be used with graphical application snaps.
+    It defines a single plug "graphics-core{base version number}" which is a content
+    interface to the "mesa-core{base version number}" snap that provides the graphics libraries and
+    other important resources.  It also configures apps to use the "opengl", "wayland", and "x11" plugs,
+    which are essential for graphical output on Ubuntu-supported platforms.
+    """
+
+    @staticmethod
+    @overrides
+    def get_supported_bases() -> Tuple[str, ...]:
+        return ("core22",)
+
+    @staticmethod
+    @overrides
+    def get_supported_confinement() -> Tuple[str, ...]:
+        return "strict", "devmode"
+
+    @staticmethod
+    @overrides
+    def is_experimental(base: Optional[str]) -> bool:
+        return False
+
+    @overrides
+    def get_root_snippet(self) -> Dict[str, Any]:
+        base = self.yaml_data["base"]
+        return {
+            "plugs": {
+                _GRAPHICS_NAME[base]: {
+                    "interface": "content",
+                    "target": "$SNAP/graphics",
+                    "default-provider": _MESA_SNAP[base],
+                },
+            },
+            "layout": {
+                "/usr/share/drirc.d": {
+                    "bind": "$SNAP/graphics/drirc.d",
+                },
+                "/usr/share/libdrm": {
+                    "bind": "$SNAP/graphics/libdrm",
+                },
+                "/usr/share/X11": {
+                    "bind": "$SNAP/graphics/X11",
+                },
+            },
+        }
+
+    @overrides
+    def get_app_snippet(self) -> Dict[str, Any]:
+        return {
+            "command-chain": [
+                "bin/graphics-launch",
+            ],
+            "plugs": ["opengl", "wayland", "x11"],
+        }
+
+    @overrides
+    def get_part_snippet(self, *, plugin_name: str) -> Dict[str, Any]:
+        return {
+            "build-packages": [
+                "libdrm-dev",
+                "libegl-dev",
+                "libgbm-dev",
+                "libgl-dev",
+                "libgles-dev",
+                "libglx-dev",
+                "libnvidia-egl-wayland-dev",
+                "libva-dev",
+                "libvdpau-dev",
+                "libvulkan-dev",
+                "libwayland-dev",
+                "libwayland-egl-backend-dev",
+                "libx11-xcb-dev",
+                "libxau-dev",
+                "libxcb-dri2-0-dev",
+                "libxcb-dri3-dev",
+                "libxcb-present-dev",
+                "libxcb-sync-dev",
+                "libxcb-xfixes0-dev",
+                "libxcb1-dev",
+                "libxdamage-dev",
+                "libxdmcp-dev",
+                "libxshmfence-dev",
+            ],
```

Comment:
I really don't think these should be part of the extension. Very little software will need these development packages. And they'll know straight away what to add to their `build-packages` when the build fails…

Reply:
I've removed the segment